src/PhpSpec/Runner/Maintainer/CollaboratorsMaintainer.php

```diff
@@ -134,6 +137,20 @@ private function generateCollaborators(CollaboratorManager $collaborators, \Refl
         }
     }
 
+    private function ensureParameterHasSupportedTypeHinting(
+        \ReflectionParameter $parameter,
+        \ReflectionFunctionAbstract $function
+    ) {
+        $isUnsupportedTypeHinting = $parameter->isArray();
+        if (version_compare(PHP_VERSION, '5.4.0', '>')) {
```

Comment:
Use ``PHP_VERSION_ID >= 50400``. OPCache can optimize such check at compile time while it cannot for a function call.